From 66a2b1707a5c10db68676f9758c65063373dec48 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Juho=20H=C3=A4m=C3=A4l=C3=A4inen?=
 <juho.hamalainen@tieto.com>
Date: Wed, 16 Oct 2013 10:54:20 +0300
Subject: [PATCH 5/9] bluetooth-device: Do not lose transport pointer after
 getting it.

When u->transport has valid pointer, it seems most code assumes it to
stay non-NULL. When profile switch fails, stop_thread() clears the
pointer and this may result in assert when acquiring the transport later
on.
---
 src/modules/bluetooth/module-bluez4-device.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/modules/bluetooth/module-bluez4-device.c b/src/modules/bluetooth/module-bluez4-device.c
index 0405f655..71257966 100644
--- a/src/modules/bluetooth/module-bluez4-device.c
+++ b/src/modules/bluetooth/module-bluez4-device.c
@@ -1978,7 +1978,8 @@ static void stop_thread(struct userdata *u) {
 
     if (u->transport) {
         bt_transport_release(u);
-        u->transport = NULL;
+        /* Do not set transport pointer to NULL. When failing to switch
+         * profile NULL u->transport would assert. */
     }
 
     if (u->sink) {
-- 
2.11.1

